clm/mlperf_logging_utils.py

```diff
@@ -153,6 +163,22 @@ def on_step_end(
         Event called at the end of a training step.
         """
         if state.global_step % (state.eval_steps) == 0 and state.global_step > 0:
+            self.mllogger.event(
+                "train_loss",
+                value=state.log_history[-1]["loss"] if state.log_history else -1,
+                metadata={
+                    "samples_count": state.log_history[-1]["step"]
+                    * self.global_batch_size
```

Comment:
It would be better to use 
```
"samples_count": state.global_step * self.global_batch_tokens
```
`global_batch_size` mean batch size only while `global_batch_tokens` equals to `global_batch_size` x `sequence_length` which is actually the number of tokens trained or `samples_count` in this one.